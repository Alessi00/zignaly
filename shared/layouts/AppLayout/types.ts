// Dependencies
import { ReactNode } from "react";

export type AppLayoutProps = {
  children: ReactNode,
  title: string;
}
