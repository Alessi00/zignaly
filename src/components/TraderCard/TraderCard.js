import React from "react";
import "./TraderCard.scss";
import { Box } from "@material-ui/core";
import TraderCardHeader from "./TraderCardHeader";
import TraderCardBody from "./TraderCardBody";

/**
 * @typedef {import("../../services/tradeApiClient.types").DailyReturn} DailyReturn
 *
 * @typedef {Object} TraderCardPropTypes
 * @property {any} fee Comission fee in %
 * @property {string} name Provider's name
 * @property {string} logoUrl Url of the provider's logo
 * @property {string} coin Coin traded by provider
 * @property {boolean} showSummary Flag to indicate if summary should be rendered.
 * @property {number} risk Return for open positions.
 * @property {Array<DailyReturn>} dailyReturns Return for closed positions on the selected period.
 * @property {string} id Provider id.
 * @property {Array<string>} exchanges Exchanges supported by provider
 * @property {boolean} isCopyTrading Flag to indicate if the provider is copy trading.
 */

/**
 * Provides a card with trader's information and stats
 *
 * @param {TraderCardPropTypes} props Component properties.
 * @returns {JSX.Element} Component JSX.
 */
const TraderCard = (props) => {
  const {
    id,
    risk,
    dailyReturns,
    showSummary,
    fee,
    name,
    logoUrl,
    coin,
    exchanges,
    isCopyTrading,
  } = props;

  return (
    <Box
      bgcolor="grid.main"
      className="traderCard"
      display="flex"
      flexDirection="column"
      justifyContent="flex-start"
    >
      <TraderCardHeader coin={coin} exchanges={exchanges} fee={fee} logoUrl={logoUrl} name={name} />
      <TraderCardBody
        dailyReturns={dailyReturns}
        id={id}
        isCopyTrading={isCopyTrading}
        risk={risk}
        showSummary={showSummary}
      />
    </Box>
  );
};

export default TraderCard;
