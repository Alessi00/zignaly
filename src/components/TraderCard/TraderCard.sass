@import '../../styles/variables.sass';

.traderCard
    box-sizing: border-box;
    width: 405px;
    min-height: 300px;
    margin: 30px 21px 30px 0px;
    border: 1px solid $borderColor;
    border-radius: 4px;

    .traderCardHeader
        width: 100%;
        padding: 0px 20px;
        height: 80px;
        border-bottom: 1px solid $borderColor;
        .logoIcon
            width: 40px;
            height: 40px;
        .traderCardHeaderTitleBox
            width: calc(100% - 60px);
            height: 40px;
            margin-left: 20px;
            .nameBox
                height: 40px;
                width: fit-content;
                .name
                    color: $purple;
                .connectedIcon
                    width: 20px;
                    height: 20px;
                    margin-left: 7px;
                    margin-top: -1px;
                .tradeType
                    font-size: 14px;
            .commissionBox
                height: 40px;
                width: fit-content;
    .traderCardBody
        .returnsBox
            padding: 0px 20px;
            height: 60px;
            .returns
                height: 40px;
                h4
                    &.green
                        color: $green;
                    &.red
                        color: $red;
            .openPositions
                height: 40px;
                h4
                    &.green
                        color: $green;
                    &.red
                        color: $red;
        .traderCardGraph
            width: 100%;
            position: relative;
            .chartCanvas
                width: 100% !important;
            .actions
                width: 250px;
                position: absolute;
                bottom: 16px;
                left: 50%;
                transform: translate(-50%, 0);
        .userSummary
            padding: 10px 20px;
            h5
                margin: 5px 0 10px 0;
                &:last-child
                    margin: 5px 0 5px 0;
                &.green
                    color: $green;
                &.red
                    color: $red;
@media only screen and (max-width: 600px)
    .traderCard
        width: 100%;
        margin: 30px 0 0 0;

        .traderCardHeader
            width: 100%;
            padding: 0px 10px;
            .logoIcon
                width: 30px;
                height: 30px;
            .traderCardHeaderTitleBox
                width: calc(100% - 40px);
                margin-left: 15px;
                .nameBox
                    height: 40px;
                    width: fit-content;
                    .name
                        color: $purple;
                        text-overflow: ellipsis !important;
                        white-space: nowrap !important;
                        overflow: hidden !important;
                .commissionBox
                    height: 40px;
                    width: fit-content;
        .traderCardBody
            .returnsBox
                padding: 0px 10px;
            .userSummary
                padding: 10px;
