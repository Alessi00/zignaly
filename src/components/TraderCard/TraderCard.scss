.traderCard {
  border: 1px solid $borderColor;
  border-radius: 4px;
  box-sizing: border-box;
  justify-content: flex-start;
  display: flex;
  flex-direction: column;
  background: $white;
}

@include media("<=sm") {
  .traderCard {
    margin: 30px 0 0;
  }
}

[data-theme="dark"] {
  .traderCard {
    background: $lightBlackBackground;
  }
}
