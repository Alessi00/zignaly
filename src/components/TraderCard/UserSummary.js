import React from "react";
import { Box, Typography } from "@material-ui/core";

const UserSummary = props => {
  const { data } = props;
  let id = "traderCard" + data;

  return (
    <Box display="flex" flexDirection="column" justifyContent="flex-start" className="userSummary">
      <Typography variant="subtitle1">Allocated</Typography>
      <Box className="returns" display="flex" flexDirection="row" justifyContent="space-between">
        <Typography variant="h5">BTC 0.2</Typography>
        <Typography variant="h5">$1280,46</Typography>
      </Box>
      <Typography variant="subtitle1">Return since copying</Typography>
      <Box className="returns" display="flex" flexDirection="row" justifyContent="space-between">
        <Typography variant="h5" className="green">
          7.2%
        </Typography>
        <Typography variant="h5" className="green">
          19%
        </Typography>
      </Box>
      <Typography variant="subtitle1">Open positions now</Typography>
      <Box className="returns" display="flex" flexDirection="row" justifyContent="space-between">
        <Typography variant="h5" className="red">
          .7%%
        </Typography>
        <Typography variant="h5" className="red">
          $2.4
        </Typography>
      </Box>
    </Box>
  );
};

export default UserSummary;
