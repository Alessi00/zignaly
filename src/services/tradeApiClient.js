import fetch from "cross-fetch";
import {
  userCreateResponseTransform,
  userEntityResponseTransform,
  userPositionsResponseTransform,
  providersResponseTransform,
  providersStatsResponseTransform,
  userExchangeConnectionResponseTransform,
  userBalanceResponseTransform,
  userPositionItemTransform,
  userEquityResponseTransform,
  quotesResponseTransform,
  basesResponseTransform,
  connectedProviderUserInfoResponseTransform,
  providerGetResponseTransform,
  serverTimeResponseTransform,
  coinRayTokenResponseTransform,
  exchangeMarketDataResponseTransform,
  exchangeListResponseTransform,
  ownCopyTraderProvidersOptionsResponseTransform,
  providerFollowersResponseTransform,
  providerFollowersListResponseTransform,
  providerPerformanceResponseTransform,
  userGetResponseTransform,
} from "./tradeApiClient.types";

/**
 * @typedef {import('./tradeApiClient.types').AuthorizationPayload} AuthorizationPayload
 * @typedef {import('./tradeApiClient.types').UserEquityPayload} UserEquityPayload
 * @typedef {import('./tradeApiClient.types').PositionActionPayload} PositionActionPayload
 * @typedef {import('./tradeApiClient.types').PositionEntity} PositionEntity
 * @typedef {import('./tradeApiClient.types').PositionsListPayload} PositionsListPayload
 * @typedef {import('./tradeApiClient.types').ProvidersCollection} ProvidersCollection
 * @typedef {import('./tradeApiClient.types').ProvidersPayload} ProvidersPayload
 * @typedef {import('./tradeApiClient.types').ProvidersStatsCollection} ProvidersStatsCollection
 * @typedef {import('./tradeApiClient.types').ProvidersStatsPayload} ProvidersStatsPayload
 * @typedef {import('./tradeApiClient.types').UserCreatePayload} UserCreatePayload
 * @typedef {import('./tradeApiClient.types').UserCreateResponse} UserCreateResponse
 * @typedef {import('./tradeApiClient.types').UserLoginPayload} UserLoginPayload
 * @typedef {import('./tradeApiClient.types').UserLoginResponse} UserLoginResponse
 * @typedef {import('./tradeApiClient.types').UserPositionsCollection} UserPositionsCollection
 * @typedef {import('./tradeApiClient.types').GetProviderPayload} GetProviderPayload
 * @typedef {import('./tradeApiClient.types').GetProviderFollowersPayload} GetProviderFollowersPayload
 * @typedef {import('./tradeApiClient.types').ConnectProviderPayload} ConnectProviderPayload
 * @typedef {import('./tradeApiClient.types').DisableProviderPayload} DisableProviderPayload
 * @typedef {import('./tradeApiClient.types').EditProvderPayload} EditProvderPayload
 * @typedef {import('./tradeApiClient.types').BaseAssetsPayload} BaseAssetsPayload
 * @typedef {import('./tradeApiClient.types').ConnectedProviderUserInfoPayload} ConnectedProviderUserInfoPayload
 * @typedef {import('./tradeApiClient.types').ConnectedProviderUserInfo} ConnectedProviderUserInfo
 * @typedef {import('./tradeApiClient.types').CoinRayToken} CoinRayToken
 * @typedef {import('./tradeApiClient.types').MarketSymbolsCollection} MarketSymbolsCollection
 * @typedef {import('./tradeApiClient.types').CopyTradersProvidersOptionsPayload} CopyTradersProvidersOptionsPayload
 * @typedef {import('./tradeApiClient.types').CopyTradersProvidersOptionsCollection} CopyTradersProvidersOptionsCollection
 * @typedef {import('./tradeApiClient.types').ExchangeAddPayload} ExchangeAddPayload
 * @typedef {import('./tradeApiClient.types').ExchangeDeletePayload} ExchangeDeletePayload
 * @typedef {import('./tradeApiClient.types').ExchangeUpdatePayload} ExchangeUpdatePayload
 */

/**
 * Trade API client service, provides integration to API endpoints.
 *
 * @constructor
 * @public
 * @class TradeApiClient
 */
class TradeApiClient {
  /**
   * Creates an instance of TradeApiClient.
   * @memberof TradeApiClient
   */
  constructor() {
    this.baseUrl = process.env.GATSBY_TRADEAPI_URL;
  }

  /**
   * Process API HTTP request.
   *
   * @param {string} endpointPath API endpoint path and action.
   * @param {Object} payload Request payload parameters object.
   * @returns {Promise<*>} Promise that resolves Trade API request response.
   *
   * @memberof TradeApiClient
   */
  async doRequest(endpointPath, payload) {
    let responseData = {};
    const requestUrl = this.baseUrl + endpointPath;
    const options = {
      method: "POST",
      body: JSON.stringify(payload),
      headers: {
        "Content-Type": "application/json",
      },
    };

    try {
      const response = await fetch(requestUrl, options);
      if (response.status === 200) {
        responseData = await response.json();
      } else {
        responseData.error = await response.json();
      }
    } catch (e) {
      responseData.error = e.message;
    }

    if (responseData.error) {
      const customError = responseData.error.error;
      throw customError;
    }

    return responseData;
  }

  /**
   * Login a user in Trade API.
   *
   * @param {UserLoginPayload} payload User login payload
   *
   * @returns {Promise<UserLoginResponse>} Promise that resolves user login response
   *
   * @memberof TradeApiClient
   */
  async userLogin(payload) {
    const endpointPath = "/fe/api.php?action=login";
    const responseData = await this.doRequest(endpointPath, payload);

    return userEntityResponseTransform(responseData);
  }

  userLogout() {}

  /**
   * Create user at Zignaly Trade API.
   *
   * @param {UserCreatePayload} payload User create payload.
   *
   * @returns {Promise<UserCreateResponse>} Promise that resolves user create response.
   *
   * @memberof TradeApiClient
   */
  async userCreate(payload) {
    const endpointPath = "/fe/api.php?action=signup";
    const responseData = await this.doRequest(endpointPath, payload);

    return userCreateResponseTransform(responseData);
  }

  /**
   * Get user open trading positions.
   *
   * @param {PositionsListPayload} payload User authorization payload.

   * @returns {Promise<UserPositionsCollection>} Promise that resolve user positions collection.
   *
   * @memberof TradeApiClient
   */
  async openPositionsGet(payload) {
    const endpointPath = "/fe/api.php?action=getOpenPositions";
    const responseData = await this.doRequest(endpointPath, payload);

    return userPositionsResponseTransform(responseData);
  }

  /**
   * Get user closed trading positions.
   *
   * @param {PositionsListPayload} payload User authorization payload.

   * @returns {Promise<UserPositionsCollection>} Promise that resolve user positions collection.
   *
   * @memberof TradeApiClient
   */
  async closedPositionsGet(payload) {
    const endpointPath = "/fe/api.php?action=getClosedPositions";
    const responseData = await this.doRequest(endpointPath, {
      type: "sold",
      ...payload,
    });

    return userPositionsResponseTransform(responseData);
  }

  /**
   * Get user unsold / unopened trading positions.
   *
   * @param {PositionsListPayload} payload User authorization payload.

   * @returns {Promise<UserPositionsCollection>} Promise that resolve user positions collection.
   *
   * @memberof TradeApiClient
   */
  async logPositionsGet(payload) {
    const endpointPath = "/fe/api.php?action=getClosedPositions";
    const responseData = await this.doRequest(endpointPath, {
      type: "allClosedExtended",
      ...payload,
    });

    return userPositionsResponseTransform(responseData);
  }

  /**
   * Get providers list.
   *
   * @param {ProvidersPayload} payload Get providers payload.

   * @returns {Promise<ProvidersCollection>} Promise that resolves providers collection.
   *
   * @memberof TradeApiClient
   */
  async providersGet(payload) {
    const endpointPath = "/fe/api.php?action=getProviderList2";
    const responseData = await this.doRequest(endpointPath, payload);

    return providersResponseTransform(responseData);
  }

  /**
   *
   *
   * @param {AuthorizationPayload} payload
   * @returns
   * @memberof TradeApiClient
   */

  async userExchangesGet(payload) {
    const endpointPath = "/fe/api.php?action=getUserExchanges";
    const responseData = await this.doRequest(endpointPath, payload);

    return userExchangeConnectionResponseTransform(responseData);
  }

  /**
   *
   *
   * @param {UserEquityPayload} payload
   * @returns
   * @memberof TradeApiClient
   */

  async userBalanceGet(payload) {
    const endpointPath = "/fe/api.php?action=getQuickExchangeSummary";
    const responseData = await this.doRequest(endpointPath, payload);

    return userBalanceResponseTransform(responseData);
  }

  /**
   *
   *
   * @param {UserEquityPayload} payload
   * @returns
   * @memberof TradeApiClient
   */

  async userEquityGet(payload) {
    const endpointPath = "/fe/api.php?action=getHistoricalBalance";
    const responseData = await this.doRequest(endpointPath, payload);

    return userEquityResponseTransform(responseData);
  }

  /**
   * Get providers profits stats.
   *
   * @param {ProvidersStatsPayload} payload Get providers stats payload.

   * @returns {Promise<ProvidersStatsCollection>} Promise that resolves providers stats collection.
   *
   * @memberof TradeApiClient
   */
  async providersStatsGet(payload) {
    const endpointPath = "/fe/api.php?action=getProviderProfitStats";
    const responseData = await this.doRequest(endpointPath, payload);

    return providersStatsResponseTransform(responseData);
  }

  /**
   * Close a position.
   *
   * @param {PositionActionPayload} payload User authorization payload.

   * @returns {Promise<PositionEntity>} Promise that resolve user affected position entity.
   *
   * @memberof TradeApiClient
   */
  async positionClose(payload) {
    const endpointPath = "/fe/api.php?action=closePosition";
    const responseData = await this.doRequest(endpointPath, payload);

    return userPositionItemTransform(responseData);
  }

  /**
   * Exit a position.
   *
   * @param {PositionActionPayload} payload User authorization payload.

   * @returns {Promise<PositionEntity>} Promise that resolve user affected position entity.
   *
   * @memberof TradeApiClient
   */
  async positionExit(payload) {
    const endpointPath = "/fe/api.php?action=sellPosition";
    const responseData = await this.doRequest(endpointPath, payload);

    return userPositionItemTransform(responseData);
  }

  /**
   * @typedef {import('./tradeApiClient.types').ReadOnlyPayload} ReadOnlyPayload
   * @typedef {import('./tradeApiClient.types').QuoteAssetsDict} QuoteAssetsDict
   * @typedef {import('./tradeApiClient.types').BaseAssetsDict} BaseAssetsDict
   */

  /**
   *
   * Get quote assets.
   *
   * @param {ReadOnlyPayload} payload
   * @returns {Promise<QuoteAssetsDict>} Promise that resolves quote assets.
   * @memberof TradeApiClient
   */

  async quotesAssetsGet(payload) {
    const endpointPath = "/fe/api.php?action=getQuoteAssets";
    const responseData = await this.doRequest(endpointPath, payload);

    return quotesResponseTransform(responseData);
  }

  /**
   * Get providers profits stats.
   *
   * @param {GetProviderPayload} payload Get providers stats payload.

   * @returns {Promise<*>} Returns promise.
   *
   * @memberof TradeApiClient
   */
  async providerGet(payload) {
    const endpointPath = "/fe/api.php?action=getProvider";
    const responseData = await this.doRequest(endpointPath, payload);

    return providerGetResponseTransform(responseData);
  }
  /**
   * @typedef {import('./tradeApiClient.types').ServerTime} ServerTime
   */

  /**
   * Get Trade API server time.
   *
   * @param {AuthorizationPayload} payload User authorization.
   * @returns {Promise<ServerTime>} Promise that resolves server time value object.
   *
   * @memberof TradeApiClient
   */
  async serverTimeGet(payload) {
    const endpointPath = "/fe/ohlc.php?action=fetchTime";
    const responseData = await this.doRequest(endpointPath, payload);

    return serverTimeResponseTransform(responseData);
  }

  /**
   * Get a coinray access token for authenticated Trade API user.
   *
   * @param {AuthorizationPayload} payload User authorization.
   * @returns {Promise<CoinRayToken>} Promise that resolves server time value object.
   *
   * @memberof TradeApiClient
   */
  async coinRayTokenGet(payload) {
    const endpointPath = "/fe/api.php?action=getCoinRayToken";
    const responseData = await this.doRequest(endpointPath, payload);

    return coinRayTokenResponseTransform(responseData);
  }

  /**
   * Get user exchange connnection market data.
   *
   * @param {AuthorizationPayload} payload Authorized exchange data payload.
   * @returns {Promise<MarketSymbolsCollection>} Promise that resolves exchange market (symbols) data collection.
   *
   * @memberof TradeApiClient
   */
  async exchangeConnectionMarketDataGet(payload) {
    const endpointPath = "/fe/api.php?action=getPairsNew";
    const responseData = await this.doRequest(endpointPath, payload);

    return exchangeMarketDataResponseTransform(responseData);
  }

  /**
   *
   * Get quote assets.
   *
   * @param {BaseAssetsPayload} payload
   * @returns {Promise<BaseAssetsDict>} Promise that resolves base assets.
   * @memberof TradeApiClient
   */

  async baseAssetsGet(payload) {
    const endpointPath = "/fe/api.php?action=getBaseAssets";
    const responseData = await this.doRequest(endpointPath, payload);

    return basesResponseTransform(responseData);
  }

  /**
   *
   * Get connected provider user info.
   *
   * @param {ConnectedProviderUserInfoPayload} payload
   * @returns {Promise<ConnectedProviderUserInfo>} Promise that resolves connected trader user info.
   * @memberof TradeApiClient
   */

  async connectedProviderUserInfoGet(payload) {
    const endpointPath = "/fe/api.php?action=getCurrentAllocatedAndProfitSinceFollowing";
    const responseData = await this.doRequest(endpointPath, payload);

    return connectedProviderUserInfoResponseTransform(responseData);
  }

  /**
   * Get providers profits stats.
   *
   * @param {ConnectProviderPayload} payload Get providers stats payload.

   * @returns {Promise<*>} Returns promise.
   *
   * @memberof TradeApiClient
   */
  async providerConnect(payload) {
    const endpointPath = "/fe/api.php?action=createProvider";
    const responseData = await this.doRequest(endpointPath, payload);

    return responseData;
  }

  /**
   * Get providers profits stats.
   *
   * @param {DisableProviderPayload} payload Get providers stats payload.

   * @returns {Promise<*>}
   *
   * @memberof TradeApiClient
   */

  async providerDisable(payload) {
    const endpointPath = "/fe/api.php?action=toggleProvider";
    const responseData = await this.doRequest(endpointPath, payload);

    return responseData;
  }

  /**
   * Get providers profits stats.
   *
   * @param {AuthorizationPayload} payload Get providers stats payload.

   * @returns {Promise<*>}
   *
   * @memberof TradeApiClient
   */

  async exchangeListGet(payload) {
    const endpointPath = "/fe/api.php?action=getExchangeList";
    const responseData = await this.doRequest(endpointPath, payload);

    return exchangeListResponseTransform(responseData);
  }
  /**
   * Get copy trader providers options.
   *
   * @param {CopyTradersProvidersOptionsPayload} payload Get own copy trader providers options payload.
   * @returns {Promise<CopyTradersProvidersOptionsCollection>} Promise that resolves own copy trader providers options.
   * @memberof TradeApiClient
   */
  async userOwnCopyTradersProvidersOptions(payload) {
    const endpointPath = "/fe/api.php?action=getCopyTradingProvidersOptions";
    const responseData = await this.doRequest(endpointPath, payload);

    return ownCopyTraderProvidersOptionsResponseTransform(responseData);
  }

  /**
   * Connect to a new exchange.
   *
   * @param {ExchangeAddPayload} payload Payload
   * @returns {Promise<boolean>} Result
   * @memberof TradeApiClient
   */
  async exchangeAdd(payload) {
    const endpointPath = "/fe/api.php?action=addExchange";
    const responseData = await this.doRequest(endpointPath, payload);
    return responseData;
  }

  /**
   * Delete an exchange.
   *
   * @param {ExchangeDeletePayload} payload Payload
   * @returns {Promise<boolean>} Result
   * @memberof TradeApiClient
   */
  async exchangeDelete(payload) {
    const endpointPath = "/fe/api.php?action=delNewExchange";
    const responseData = await this.doRequest(endpointPath, payload);
    return responseData;
  }

  /**
   * Update an exchange.
   *
   * @param {ExchangeUpdatePayload} payload Payload
   * @returns {Promise<boolean>} Result
   * @memberof TradeApiClient
   */
  async exchangeUpdate(payload) {
    const endpointPath = "/fe/api.php?action=updateNewExchange";
    const responseData = await this.doRequest(endpointPath, payload);
    return responseData;
  }

  /**
   * Get providers profits stats.
   *
   * @param {EditProvderPayload} payload Get providers stats payload.

   * @returns {Promise<*>} Returns promise.
   *
   * @memberof TradeApiClient
   */
  async providerEdit(payload) {
    const endpointPath = "/fe/api.php?action=editProvider";
    const responseData = await this.doRequest(endpointPath, payload);

    return responseData;
  }

  /**
   * Get providers profits stats.
   *
   * @param {GetProviderFollowersPayload} payload Get providers stats payload.

   * @returns {Promise<*>} Returns promise.
   *
   * @memberof TradeApiClient
   */
  async providerFollowersGet(payload) {
    const endpointPath = "/fe/api.php?action=getFollowersChartForProvider";
    const responseData = await this.doRequest(endpointPath, payload);

    return providerFollowersResponseTransform(responseData);
  }

  /**
   * Get providers profits stats.
   *
   * @param {GetProviderFollowersPayload} payload Get providers stats payload.

   * @returns {Promise<*>} Returns promise.
   *
   * @memberof TradeApiClient
   */
  async providerFollowersListGet(payload) {
    const endpointPath = "/fe/api.php?action=getFollowersList";
    const responseData = await this.doRequest(endpointPath, payload);

    return providerFollowersListResponseTransform(responseData);
  }

  /**
   * Get providers profits stats.
   *
   * @param {GetProviderFollowersPayload} payload Get providers stats payload.

   * @returns {Promise<*>} Returns promise.
   *
   * @memberof TradeApiClient
   */
  async providerPerformanceGet(payload) {
    const endpointPath = "/fe/api.php?action=getProviderPerformance";
    const responseData = await this.doRequest(endpointPath, payload);

    return providerPerformanceResponseTransform(responseData);
  }

  /**
   * Get providers profits stats.
   *
   * @param {AuthorizationPayload} payload Get providers stats payload.

   * @returns {Promise<*>} Returns promise.
   *
   * @memberof TradeApiClient
   */
  async userDataGet(payload) {
    const endpointPath = "/fe/api.php?action=getUserData";
    const responseData = await this.doRequest(endpointPath, payload);

    return userGetResponseTransform(responseData);
  }
}

// JS export by default guarantee a singleton instance if we export the class
// instance, see:
// https://medium.com/@lazlojuly/are-node-js-modules-singletons-764ae97519af
const client = new TradeApiClient();
Object.freeze(client);

export default client;
