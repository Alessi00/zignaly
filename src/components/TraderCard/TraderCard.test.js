import React from "react";
import TraderCard from "./TraderCard";
import { render } from "test-utils";
import providers from "utils/test/fixtures/providers";
import { providersResponseTransform } from "services/tradeApiClient.types";

it("renders correctly", () => {
  const providersTransformed = providersResponseTransform(providers);
  const tree = render(
    <TraderCard provider={providersTransformed[0]} showSummary={true} timeFrame={30} />,
  );
  expect(tree).toMatchSnapshot();
});
