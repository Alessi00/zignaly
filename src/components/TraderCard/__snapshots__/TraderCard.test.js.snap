// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="traderCard"
      >
        <div
          class="traderCardHeader"
        >
          <div
            class="lazyload-wrapper"
          >
            <img
              class="providerLogo"
              height="40px"
              src="test-file-stub"
              title="Darkcryptolord"
              width="40px"
            />
          </div>
          <div
            class="traderCardHeaderTitleBox"
          >
            <div
              class="nameBox"
            >
              <a
                class="name"
                href="/copyTraders/5edf6165fc869626686e8657"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4"
                >
                  Darkcryptolord
                </h4>
              </a>
              <div
                class="commissionBox"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4"
                >
                  <span>
                    50
                    srv.pricemonth
                  </span>
                </h4>
                <h6
                  class="MuiTypography-root price MuiTypography-subtitle1"
                >
                  srv.edit.price
                </h6>
              </div>
            </div>
            <div
              class="tradesInfoBox"
            >
              <span
                class="MuiTypography-root tradeType MuiTypography-caption"
              >
                srv.trades
              </span>
              <img
                alt="binance"
                class="exchangeIcon binance xsmall "
                src="test-file-stub"
                title="binance"
              />
              <img
                alt="zignaly"
                class="exchangeIcon zignaly xsmall "
                src="test-file-stub"
                title="zignaly"
              />
            </div>
          </div>
        </div>
        <div
          class="lazyload-wrapper"
        >
          <div
            class="lazyload-placeholder"
            style="height: 310px;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="traderCard"
    >
      <div
        class="traderCardHeader"
      >
        <div
          class="lazyload-wrapper"
        >
          <img
            class="providerLogo"
            height="40px"
            src="test-file-stub"
            title="Darkcryptolord"
            width="40px"
          />
        </div>
        <div
          class="traderCardHeaderTitleBox"
        >
          <div
            class="nameBox"
          >
            <a
              class="name"
              href="/copyTraders/5edf6165fc869626686e8657"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                Darkcryptolord
              </h4>
            </a>
            <div
              class="commissionBox"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                <span>
                  50
                  srv.pricemonth
                </span>
              </h4>
              <h6
                class="MuiTypography-root price MuiTypography-subtitle1"
              >
                srv.edit.price
              </h6>
            </div>
          </div>
          <div
            class="tradesInfoBox"
          >
            <span
              class="MuiTypography-root tradeType MuiTypography-caption"
            >
              srv.trades
            </span>
            <img
              alt="binance"
              class="exchangeIcon binance xsmall "
              src="test-file-stub"
              title="binance"
            />
            <img
              alt="zignaly"
              class="exchangeIcon zignaly xsmall "
              src="test-file-stub"
              title="zignaly"
            />
          </div>
        </div>
      </div>
      <div
        class="lazyload-wrapper"
      >
        <div
          class="lazyload-placeholder"
          style="height: 310px;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
